scripts/onboarding/managed/upgrade-monitoring.sh

```diff
@@ -281,11 +281,26 @@ set_azure_subscription() {
   echo "successfully configured subscription id: ${subscriptionId} as current subscription for the azure cli"
 }
 
+validate_and_configure_supported_cloud() {
+  echo "get active azure cloud name configured to azure cli"
+  azureCloudName=$(az cloud show --query name -o tsv | tr "[:upper:]" "[:lower:]")
+  echo "active azure cloud name configured to azure cli: ${azureCloudName}"
+  if [ "$isArcK8sCluster" = true ]; then
+    if [ "$azureCloudName" != "azurecloud" -a  "$azureCloudName" != "azureusgovernment" ]; then
+      echo "-e only supported clouds are AzureCloud and AzureUSGovernment for Azure Arc enabled Kubernetes cluster type"
+      exit 1
+    fi
+  else
+    # For ARO v4, only supported cloud is public so just configure to public to keep the existing behavior
+    configure_to_public_cloud
```

Comment:
I see this is keeping existing behavior, but this will only work for non-ARC  clusters in public cloud?